filecoin/lotus.go

```diff
@@ -45,7 +45,10 @@ func NewLotusRPC(ctx context.Context, addr string, header http.Header) (API, err
 		header,
 	)
 	res.closer = closer
-	return &res, err
+	if err != nil {
+		return nil, err
+	}
```

Comment:
shouldn't this be before the `res.closer = closer` ?

Reply:
Yeah you're right I thought we were calling it somewhere so I chose to set it but we can't be calling it actually